crates/fluvio-protocol-derive/src/util.rs

```diff
@@ -1,75 +1,110 @@
+    {
+        match &item {
+            Meta::NameValue(kv) => {
+                if kv.path.is_ident(attr_name) {
+                    result_meta = Some(kv.clone());
+                    break;
+                }
+            }
+            _ => {}
+        };
+    }
 
-pub(crate) fn find_name_attribute<'a>(meta: &'a Meta, name: &str) -> Option<&'a MetaNameValue> {
-    find_meta(meta, name).map(|meta| match meta {
-        Meta::NameValue(name_value) => name_value,
-        _ => panic!("should not happen"),
-    })
+    result_meta
 }
 
-pub(crate) fn find_meta<'a>(meta: &'a Meta, name: &str) -> Option<&'a Meta> {
+pub(crate) fn find_name_value_from_meta(meta: &Meta, attr_name: &str) -> Option<MetaNameValue> {
+    let mut result_meta: Option<MetaNameValue> = None;
     if let Meta::List(list) = meta {
-        for attr in list.nested.iter() {
-            if let NestedMeta::Meta(named_meta) = attr {
-                match named_meta {
-                    Meta::NameValue(meta_name_value) => {
-                        if meta_name_value.path.is_ident(name) {
-                            return Some(named_meta);
-                        }
-                    }
-                    Meta::Path(path) => {
-                        if path.is_ident(name) {
-                            return Some(named_meta);
-                        }
-                    }
-                    Meta::List(_) => {}
-                }
-            }
-        }
+        result_meta = find_name_value_from_meta_list(list, attr_name);
     }
 
-    None
+    result_meta
 }
 
-/// find name value with integer value
-pub(crate) fn find_int_name_value(version_meta: &Meta, attr_name: &str) -> Option<u64> {
-    if let Some(attr) = find_name_attribute(version_meta, attr_name) {
-        match &attr.lit {
-            Lit::Int(version_val) => {
-                //  println!("version value: {}",version_val.value());
-                version_val.base10_parse::<u64>().ok()
-            }
-            _ => unimplemented!(),
+/// find name value with str value
+pub(crate) fn find_string_name_value(meta: &Meta, attr_name: &str) -> Option<LitStr> {
+    let mut value: Option<LitStr> = None;
+    if let Some(meta_name_value) = find_name_value_from_meta(meta, attr_name) {
+        let data = get_lit_str(String::from(attr_name), &meta_name_value.value);
+        value = match data {
+            Ok(val) => Some(val.clone()),
+            Err(_) => None,
         }
-    } else {
-        None
     }
+
+    value
 }
 
-/// find name value with str value
-pub(crate) fn find_string_name_value(version_meta: &Meta, attr_name: &str) -> Option<LitStr> {
-    if let Some(attr) = find_name_attribute(version_meta, attr_name) {
-        match &attr.lit {
-            Lit::Str(val) => Some(val.clone()),
-            _ => unimplemented!(),
+pub(crate) fn find_int_name_value(meta: &Meta, attr_name: &str) -> Option<u64> {
+    let mut value: Option<u64> = None;
```

Comment:
I would recommend avoiding mutable variables where it's possible. See some similar suggestions above.